File: backend/dotnet/Models/ActivityHistory.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json;

namespace dotnet.Models
{
    [Table("activity_history")]
    public class ActivityHistory
    {
        [Key]
        [Column("history_id")]
        public string Id { get; set; } = string.Empty;

        [Column("user_id")]
        public string UserId { get; set; } = string.Empty;

        [Column("description")]
        public string Description { get; set; } = string.Empty;

        [Column("done_at")]
        public DateTime DoneAt { get; set; } = DateTime.UtcNow;

        // Navigation property
        public User User { get; set; } = null!;

    }
}

File: backend/springboot/src/main/java/tn/maiko26/springboot/models/ActivityHistory.java
package tn.maiko26.springboot.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.Date;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Data
@Table(name = "activity_history")
public class ActivityHistory {
    @Id
    @Column(name = "history_id")
    private String id;

    @Column(name = "description", nullable = false)
    private String description;

    @Column(name = "done_at", nullable = false, columnDefinition = "timestamp default now()")
    private Date doneAt;

    // Relations

    @OneToOne
    @JoinColumn(name = "user_id", referencedColumnName = "email")
    private User user;
}
